sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -1147,6 +1147,18 @@ public interface OptionsConfiguration<T extends SentryOptions> {
     return getCurrentScopes().continueTrace(sentryTrace, baggageHeaders);
   }
 
+  /**
+   * Continue a trace based on the trace ID and span ID provided
+   *
+   * @param traceId the trace ID
+   * @param spanId the span ID
+   */
+  // return TransactionContext (if performance enabled) or null (if performance disabled)
+  public static void setTrace(
+      final @NotNull String traceId, final @NotNull String spanId) {
+    getCurrentScopes().setTrace(traceId, spanId);
+  }
```

Comment:
This looks good and could be used in RN as is.

Rn will propagate the `traceId` from the JS scope.

https://github.com/getsentry/sentry-javascript/blob/fc48f8397d1ed647a4e2d95dba33f022df03daa3/packages/core/src/scope.ts#L120